images/ipfailover/keepalived/lib/utils.sh

```diff
@@ -21,6 +21,154 @@ function scrub() {
   echo "${val//-/_}"
 }
 
+#
+#  Tests if an IPv4 address is valid
+#  Echos true (0) if valid, false (1) if invalid 
+#
+#  Examples:
+#     validate_ipv4 192.0.2.3
+#         # -> 0 
+#
+#     validate_ipv4 192.0.3.4.0 
+#         # -> 1 
+#
+#     validate_ipv4 192.0.2 
+#         # -> 1 
+#
+function validate_ipv4() {
+  local IPv4_GROUP="[0-2]?[0-9]{1,2}"
+  local IPv4_SHAPE="^(${IPv4_GROUP}\.){3,3}(${IPv4_GROUP})$"
+
+  if [[ ${1} =~ ${IPv4_SHAPE} ]]; then
+    local is_valid=0
+    for i in $(echo ${1} | tr "." " "); do
+      if [ $i -gt 255 ]; then
+        local is_valid=1
+      fi
+    done
+  else
+    local is_valid=1
```

Comment:
Yes, that's due to "true" being 0 in bash, so is_valid=1 is actually is_valid=false. As we're dealing with function returns, not echoes, I found it best to stick to the convention.